crates/server/README.md

```diff
@@ -0,0 +1,257 @@
+The `Query` method retrieves data from the applications running on the Calimero nodes. For instance, in the Only Peers forum application, posts and comments stored in the application's storage can be queried using the JSON-RPC interface. This enables users to fetch and display content from the forum.
+
+#### Mutate Method
+The `Mutate` method allows modification of the application's data. For example, in the Only Peers forum application, users can create new posts or comments. The Mutate method updates the application's storage with these new entries, facilitating dynamic content creation and interaction within the application.
+
+The JSON-RPC component is crucial for enabling interactive and real-time data management between P2P application front-ends and the back-end contexts running on Calimero nodes.
+
+### 3. Websocket
+
+The WebSocket is used for subscribing to and unsubscribing from certain contexts within the Calimero server. Defined handlers manage subscription states for WebSocket connections, allowing clients to receive updates about specific contexts they are interested in.
+WebSocket handlers are essential for managing real-time, context-specific subscriptions within the Calimero server. They allow clients to dynamically subscribe to and unsubscribe from updates about various application contexts, enhancing the interactivity and responsiveness of the P2P application.
+
+
+#### Subscription Handling:
+
+Websocket handles requests to subscribe to specific contexts and send responses back to the client with the subscribed context IDs.
+
+#### Unsubscription Handling:
+
+Websocket handle requests to unsubscribe from specific contexts and send responses back to the client with the unsubscribed context IDs.
+
+## Calimero Server Workflows
+
+### Workflow for Admin Dashboard
+
+```mermaid
+flowchart TD
+    A[Admin Dashboard] -->|HTTP Requests| B[Admin API]
+    B -->|HTTP Responses| A
+    B -->|Node Communication| D[Node]
+    D -->|Response| B
+
+    subgraph Server
+        B
+    end
+```
+
+### Workflow for P2P Web applications and JSON rpc
+
+```mermaid
+flowchart TD
+    A[P2P Web Application] -->|HTTP Auth Request| B[Admin API]
+    B -->|Auth Response| A
+    A -->|JSON-RPC Query/Mutate| C[JSON rpc]
+    C -->|Query/Mutate| D[Node]
+    D -->|Response| C
+    C -->|Response| A
+
+    subgraph Server
+        B
+        C
+    end
+```
+
+### Workflow for P2P Web application and Websocket
+```mermaid
+flowchart TD
+    A[Admin Dashboard] -->|HTTP Subscribe/Unsubscribe Requests| B[Websocket]
+    B -->|HTTP Websocket Responses| A
+    B -->|Subscribe / Unsubscribe| D[Node]
+    D -->|Response| B
+
+    subgraph Server
+        B
+    end
+```
+
+## Admin API endpoints
+
+The Admin API endpoints are split into protected and unprotected routes, where protected routes require authentication.
+
+**Base path**: `/admin-api`
+
+### Protected Routes
+
+These routes require authentication and are protected by the AuthSignatureLayer.
```

Comment:
`AuthSignatureLayer` is not common for people to know what it is. 

`These routes require authentication using auth headers`. Then describe how it looks like and how to generate one.

Reply:
Updated and added explanations for auth headers